### PR TITLE
Change remote upgrading agents via RESTful API documentation

### PR DESCRIPTION
**Pull request by mcarmona99**
Hi team,

This PR is related to the issue https://github.com/wazuh/wazuh/issues/5536. This issue changed some API endpoints in order to adapt them to the new upgrade module.

In this PR, I have changed the steps for upgrading agents via the RESTful API. Now these steps and examples are updated according to the new upgrade module.

Regards,
Manuel.

### CODE REVIEW

**Review by AdriiiPRodri — APPROVED** [review]

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
We need to add the appropiate changes to Migration from 3.X for these endpoints in version 4.1.

File: source/user-manual/api/equivalence.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@
 Migrating from the Wazuh API 3.X
 ================================
 
-Wazuh API 4.0 introduces several new endpoints and also modifies or removes some of the old ones. The biggest change for all existing endpoints is the new response format. Endpoint responses have been changed according to the new RBAC standard and will no longer have ``items`` and ``totalitems`` fields. Most responses will have the following structure instead:
+Wazuh API 4.1 introduces several new endpoints and also modifies or removes some of the old ones. The biggest change for all existing endpoints is the new response format. Endpoint responses have been changed according to the new RBAC standard and will no longer have ``items`` and ``totalitems`` fields. Most responses will have the following structure instead:
```
Comment (davidjiglesias) [inline source/user-manual/api/equivalence.rst:9]:
```suggestion
Wazuh API v4 introduces several new endpoints and also modifies or removes some of the old ones. The biggest change for all existing endpoints is the new response format. Endpoint responses have been changed according to the new RBAC standard and will no longer have ``items`` and ``totalitems`` fields. Most responses will have the following structure instead:
```

File: source/user-manual/api/equivalence.rst [hunk 2]
```diff
@@ -26,7 +26,7 @@ Wazuh API 4.0 introduces several new endpoints and also modifies or removes some
 Migrating users
 -----------------
 
-The Wazuh API users are not migrated when upgrading the Wazuh API from 3.X to 4.0. There are numerous security changes to the Wazuh API in 4.0 and it is not advisable.
+The Wazuh API users are not migrated when upgrading the Wazuh API from 3.X to 4.1. There are numerous security changes to the Wazuh API in 4.1 and it is not advisable.
```
Comment (davidjiglesias) [inline source/user-manual/api/equivalence.rst:29]:
```suggestion
The Wazuh API users are not migrated when upgrading the Wazuh API from 3.X to 4.1. There are numerous security changes to the Wazuh API and it is not advisable.
```

File: source/user-manual/api/equivalence.rst [hunk 3]
```diff
@@ -88,15 +88,15 @@ However, using the following API calls it is easy to create new users and assign
 Equivalence table
 -----------------
 
-Wazuh API 4.0 introduces several changes to existing endpoints. It is therefore important to review those changes when migrating to ``4.0``.
+Wazuh API 4.1 introduces several changes to existing endpoints. It is therefore important to review those changes when migrating to ``4.1``.
```
Comment (davidjiglesias) [inline source/user-manual/api/equivalence.rst:91]:
```suggestion
Wazuh API v4 introduced several changes to existing endpoints. It is therefore important to review those changes when migrating to ``4.1``.
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: source/user-manual/agents/remote-upgrading/upgrading-agent.rst [hunk 4]
```diff
@@ -99,52 +102,106 @@ Using the RESTful API
...
         }
 
 
-3. Check the upgrade result using endpoint :api-ref:`GET /agents/{agent_id}/upgrade_result <operation/api.controllers.agents_controller.get_agent_upgrade>`:
+3. Check the upgrade results using endpoint :api-ref:`GET /agents/upgrade_result <operation/api.controllers.agents_controller.get_agent_upgrade>`:
 
     .. code-block:: console
 
-        # curl -k -X GET "https://localhost:55000/agents/002/upgrade_result?pretty=true" -H  "Authorization: Bearer $TOKEN"
+        # curl -k -X GET "http://localhost:55000/agents/upgrade_result?agents_list=002,003&pretty=true" -H  "Authorization: Bearer $TOKEN"
```
Comment (davidjiglesias) [inline source/user-manual/agents/remote-upgrading/upgrading-agent.rst:140]:
```suggestion
        # curl -k -X GET "https://localhost:55000/agents/upgrade_result?agents_list=002,003&pretty=true" -H  "Authorization: Bearer $TOKEN"
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
Add changes in endpoints links for RBAC reference

**Review by davidjiglesias — APPROVED** [review]

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Pull request merged**